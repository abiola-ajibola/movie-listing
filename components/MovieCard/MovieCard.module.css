.cardWrapper {
  width: 60vw;
  font-size: 1.2rem;
  border-radius: 1rem;
  margin-bottom: 2rem;
  padding: 0 20px;
  background-color: white;
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1), 0 1px 3px rgba(0, 0, 0, 0.06);
}

.cardWrapper:hover {
  box-shadow: 0 3px 10px darkgrey;
}

.card {
  display: flex;
}

.imageWrapper {
  display: flex;
  align-items: center;
  justify-content: center;
}

.image {
  display: block;
}

.detailsWrapper {
  display: flex;
}

.cardItemsWrapper {
  list-style: none;
}

.starWrapper {
  text-align: right;
}

.title {
  font-size: 1.5rem;
  font-weight: bold;
  margin-bottom: 0.5rem;
}

.rating {
  color: #ffbe0b;
  font-weight: 800;
}

@media (max-width: 720px) {
  .cardWrapper {
    width: 80vw;
  }
  .card {
    flex-direction: column;
  }
  .cardItemsWrapper {
    padding-left: 0;
  }
}

@media (max-width: 600px) {
  .cardWrapper {
    width: 90vw;
  }
}
