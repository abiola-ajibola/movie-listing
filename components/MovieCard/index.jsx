import { useState } from "react";
import Image from "next/image";
import FavoriteIcon from "../FavouriteIcon";
import styles from "./MovieCard.module.css";

export default function MovieCard({
  imageUrl,
  title,
  rating,
  date,
  id,
  onSwitch,
  favorite = false,
}) {
  const listingBaseUrl = "https://www.themoviedb.org/movie";
  const year = date.split("-")[0];
  const titleColor = favorite ? "lightray" : "darkgray";
  const valueColor = favorite ? "aliceblue" : "black";
  const handleSwitch = () => {
    onSwitch();
  };

  return (
    <div
      className={styles.cardWrapper}
      style={{ backgroundColor: favorite ? "darkblue" : "transparent" }}
    >
      <div className={styles.starWrapper}>
        <FavoriteIcon favorite={favorite} onSwitch={handleSwitch} />
      </div>
      <a href={`${listingBaseUrl}/${id}`}>
        <div className={styles.card}>
          <div className={styles.imageWrapper}>
            <Image
              width="300px"
              height="169px"
              src={imageUrl}
              alt={title}
              className={styles.image}
            />
          </div>
          <div className={styles.detailsWrapper}>
            <ul className={styles.cardItemsWrapper}>
              <li>
                <h3 className={styles.title} style={{ color: titleColor }}>
                  Title
                </h3>
                <p className={styles.value} style={{ color: valueColor }} data-testid="title-text">
                  {title}
                </p>
              </li>
              <li>
                <h3 className={styles.title} style={{ color: titleColor }}>
                  Current Rating
                </h3>
                <p className={styles.value} style={{ color: valueColor }}>
                  {rating}
                </p>
              </li>
              <li>
                <h3 className={styles.title} style={{ color: titleColor }}>
                  Year
                </h3>
                <p className={styles.value} style={{ color: valueColor }}>
                  {year}
                </p>
              </li>
            </ul>
          </div>
        </div>
      </a>
    </div>
  );
}
