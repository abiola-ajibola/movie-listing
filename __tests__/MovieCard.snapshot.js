import { create, act } from "react-test-renderer";
import MovieCard from "@/components/MovieCard";

const imageUrl =
    "https://image.tmdb.org/t/p/w300/1EAxNqdkVnp48a7NUuNBHGflowM.jpg",
  title = "Evangelion: 3.0+1.0 Thrice Upon a Time",
  rating = 8.7,
  date = "2021-03-08",
  id = 283566,
  onSwitch = () => console.log("Switch"),
  favorite = false;

it("it renders a movie card", () => {
  let component = create(
    <MovieCard
      imageUrl={imageUrl}
      title={title}
      rating={rating}
      date={date}
      id={id}
      onSwitch={onSwitch}
      favorite={favorite}
    />
  );

  expect(component.toJSON()).toMatchSnapshot();

  // update with some different props
  act(() => {
    component.update(
      <MovieCard
        imageUrl={imageUrl}
        title={title}
        rating={rating}
        date={date}
        id={id}
        onSwitch={onSwitch}
        favorite={true}
      />
    );
  });

  expect(component.toJSON()).toMatchSnapshot();
});
