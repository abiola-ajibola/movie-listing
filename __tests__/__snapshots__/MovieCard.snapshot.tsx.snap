// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a movie card 1`] = `
<div
  className="cardWrapper"
  style={
    Object {
      "backgroundColor": "transparent",
    }
  }
>
  <div
    className="starWrapper"
  >
    <span
      className="star"
      onClick={[Function]}
      style={
        Object {
          "color": "black",
        }
      }
    >
      ★
    </span>
  </div>
  <a
    href="https://www.themoviedb.org/movie/283566"
  >
    <div
      className="card"
    >
      <div
        className="imageWrapper"
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "width": "initial",
            }
          }
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27300%27%20height=%27169%27/%3e"
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            />
          </span>
          <img
            alt="Evangelion: 3.0+1.0 Thrice Upon a Time"
            className="image"
            data-nimg="intrinsic"
            decoding="async"
            onError={[Function]}
            onLoad={[Function]}
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
          <noscript>
            <img
              alt="Evangelion: 3.0+1.0 Thrice Upon a Time"
              className="image"
              data-nimg="intrinsic"
              decoding="async"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw300%2F1EAxNqdkVnp48a7NUuNBHGflowM.jpg&w=640&q=75"
              srcSet="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw300%2F1EAxNqdkVnp48a7NUuNBHGflowM.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw300%2F1EAxNqdkVnp48a7NUuNBHGflowM.jpg&w=640&q=75 2x"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
        </span>
      </div>
      <div
        className="detailsWrapper"
      >
        <ul
          className="cardItemsWrapper"
        >
          <li>
            <h3
              className="title"
              style={
                Object {
                  "color": "darkgray",
                }
              }
            >
              Title
            </h3>
            <p
              className="value"
              data-testid="title-text"
              style={
                Object {
                  "color": "black",
                }
              }
            >
              Evangelion: 3.0+1.0 Thrice Upon a Time
            </p>
          </li>
          <li>
            <h3
              className="title"
              style={
                Object {
                  "color": "darkgray",
                }
              }
            >
              Current Rating
            </h3>
            <p
              className="value"
              style={
                Object {
                  "color": "black",
                }
              }
            >
              8.7
            </p>
          </li>
          <li>
            <h3
              className="title"
              style={
                Object {
                  "color": "darkgray",
                }
              }
            >
              Year
            </h3>
            <p
              className="value"
              style={
                Object {
                  "color": "black",
                }
              }
            >
              2021
            </p>
          </li>
        </ul>
      </div>
    </div>
  </a>
</div>
`;

exports[`it renders a movie card 2`] = `
<div
  className="cardWrapper"
  style={
    Object {
      "backgroundColor": "darkblue",
    }
  }
>
  <div
    className="starWrapper"
  >
    <span
      className="star"
      onClick={[Function]}
      style={
        Object {
          "color": "yellow",
        }
      }
    >
      ★
    </span>
  </div>
  <a
    href="https://www.themoviedb.org/movie/283566"
  >
    <div
      className="card"
    >
      <div
        className="imageWrapper"
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "width": "initial",
            }
          }
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27300%27%20height=%27169%27/%3e"
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            />
          </span>
          <img
            alt="Evangelion: 3.0+1.0 Thrice Upon a Time"
            className="image"
            data-nimg="intrinsic"
            decoding="async"
            onError={[Function]}
            onLoad={[Function]}
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
          <noscript>
            <img
              alt="Evangelion: 3.0+1.0 Thrice Upon a Time"
              className="image"
              data-nimg="intrinsic"
              decoding="async"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw300%2F1EAxNqdkVnp48a7NUuNBHGflowM.jpg&w=640&q=75"
              srcSet="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw300%2F1EAxNqdkVnp48a7NUuNBHGflowM.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw300%2F1EAxNqdkVnp48a7NUuNBHGflowM.jpg&w=640&q=75 2x"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
        </span>
      </div>
      <div
        className="detailsWrapper"
      >
        <ul
          className="cardItemsWrapper"
        >
          <li>
            <h3
              className="title"
              style={
                Object {
                  "color": "lightray",
                }
              }
            >
              Title
            </h3>
            <p
              className="value"
              data-testid="title-text"
              style={
                Object {
                  "color": "aliceblue",
                }
              }
            >
              Evangelion: 3.0+1.0 Thrice Upon a Time
            </p>
          </li>
          <li>
            <h3
              className="title"
              style={
                Object {
                  "color": "lightray",
                }
              }
            >
              Current Rating
            </h3>
            <p
              className="value"
              style={
                Object {
                  "color": "aliceblue",
                }
              }
            >
              8.7
            </p>
          </li>
          <li>
            <h3
              className="title"
              style={
                Object {
                  "color": "lightray",
                }
              }
            >
              Year
            </h3>
            <p
              className="value"
              style={
                Object {
                  "color": "aliceblue",
                }
              }
            >
              2021
            </p>
          </li>
        </ul>
      </div>
    </div>
  </a>
</div>
`;
